utils/staking-miner/src/dry_run.rs

```diff
@@ -115,13 +115,23 @@ macro_rules! dry_run_cmd_for { ($runtime:ident) => { paste::paste! {
 		signer: Signer,
 	) -> Result<(), Error<$crate::[<$runtime _runtime_exports>]::Runtime>> {
 		use $crate::[<$runtime _runtime_exports>]::*;
-		let mut ext = crate::create_election_ext::<Runtime, Block>(
-			rpc.clone(),
-			config.at,
-			vec!["Staking".to_string(), "System".to_string()],
-		).await?;
-		force_create_snapshot::<Runtime>(&mut ext)?;
+		let mut ext = if config.force_snapshot {
+			let mut ext = crate::create_election_ext::<Runtime, Block>(
+				rpc.clone(),
+				config.at,
+				vec!["Staking".to_string(), "BagsList".to_string()],
```

Comment:
Don't we need "Staking".to_string()?

Reply:
correct, and we have it? I think you meant in the `else` branch? there we don't need it since we assume that snapshot already exists in the `EPM` pallet.

Reply:
you could do (to get rid of some boiler plate):

```rust
let pallets = if config.force_snapshot { vec!["Staking".to_string(), "BagsList".to_string()] } else { vec![] };
```